docs/reference/ingest/apis/simulate-ingest.asciidoc

```diff
@@ -0,0 +1,358 @@
+
+[[simulate-ingest-api]]
+=== Simulate ingest API
+++++
+<titleabbrev>Simulate ingest</titleabbrev>
+++++
+
+Executes ingest pipelines against a set of provided documents, optionally
+with substitute pipeline definitions.
+
+////
+[source,console]
+----
+PUT /_ingest/pipeline/my-pipeline
+{
+  "description" : "example pipeline to simulate",
+      "processors": [
+      {
+        "set" : {
+          "field" : "field1",
+          "value" : "value1"
+        }
+      }
+    ]
+}
+
+PUT /_ingest/pipeline/my-final-pipeline
+{
+  "description" : "example final pipeline to simulate",
+      "processors": [
+      {
+        "set" : {
+          "field" : "field2",
+          "value" : "value2"
+        }
+      }
+    ]
+}
+
+PUT /index
+{
+  "settings": {
+    "index": {
+      "default_pipeline": "my-pipeline",
+      "final_pipeline": "my-final-pipeline"
+    }
+  }
+}
+----
+// TESTSETUP
+////
+
+[source,console]
+----
+POST /_ingest/_simulate
+{
+  "docs": [
+    {
+      "_index": "index",
+      "_id": "id",
+      "_source": {
+        "foo": "bar"
+      }
+    },
+    {
+      "_index": "index",
+      "_id": "id",
+      "_source": {
+        "foo": "rab"
+      }
+    }
+  ],
+  "pipeline_substitutions": { <1>
+    "my-pipeline": {
+      "processors": [
+        {
+          "set": {
+            "field": "field3",
+            "value": "value3"
+          }
+        }
+      ]
+    }
+  }
+}
+----
+
+<1> This replaces the existing `my-pipeline` pipeline with the contents given here for the duration of this request.
+
+[[simulate-ingest-api-request]]
+==== {api-request-title}
+
+`POST /_ingest/_simulate`
+
+`GET /_ingest/_simulate`
+
+`POST /_ingest/<target>/_simulate`
+
+`GET /_ingest/<target>/_simulate`
+
+[[simulate-ingest-api-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the
+`index` or `create` <<privileges-list-indices,index privileges>>
+to use this API.
+
+[[simulate-ingest-api-desc]]
+==== {api-description-title}
+
+The simulate ingest API simulates ingesting data into an index. It
+executes the default and final pipeline for that index against a set
+of documents provided in the body of the request. If a pipeline
+contains a reroute processor, it follows that reroute processor to the
+new index, executing that index's pipelines as well the same way that
+a non-simulated ingest would. No data is indexed into ${es}. Instead,
```

Comment:
Shows up as $Elasticsearch in the actual page :)

Reply:
Is that not how we're branding this?